### DISCUSSION TITLE
Is it possible to use LeastLoad strategy with a Balancer that is using Portals?

### DISCUSSION THREAD

**Discussion by PatrickstarWritesCode** [XTLS/Xray-core#3138] [General]
This is my balancer, As soon as I use it as BalancerTag in routing rules it stops working, same balancer with Random or RoundRobin type works without issues.

```
    "balancers": [
      {
        "tag": "test",
        "selector": [
          "Portal-1",
          "Portal-2",
          "Portal-3",
          "Portal-4"
        ],
        "strategy": {
          "type": "leastLoad"
        }
      }
```

**Comment by SatenRuiko-Lv0** [comment]
you should 配置 (burst)observatory if you use leastping or leastload.
for example
"balancers": [{
			"tag": "global-balancers",
			"selector": ["global-out"],
			"strategy": {
				"type": "leastLoad"
			},
			"fallbackTag": "global-out0"
		}, {
			"tag": "cn-balancers",
			"selector": ["cn-out"],
			"strategy": {
				"type": "leastLoad"
			},
			"fallbackTag": "cn-out0"
		}]
	},
	"burstObservatory": {
		"subjectSelector": ["global-out", "cn-out", "global-udp-out"],
		"pingConfig": {
            "destination": "http://www.apple.com/library/test/success.html",
            "interval": "1h",
            "connectivity": "http://connectivitycheck.platform.hicloud.com/generate_204",
            "timeout": "10s",
            "sampling": 2
至于leastPing,also 需要配置有关的内容。而 random and RoundRobin 是no need (burst)observatory的

**Reply by PatrickstarWritesCode** [reply]
Thanks for the input this helped a lot, this raises another question for me, what is observatory used for then?

**Reply by SatenRuiko-Lv0** [reply]
怎么说呢，固定配置吧？如果你用leastPing或leastLoad，不配置observatory有关字段的话，就无法观测同一策略组中不同节点的ping或者是rtt情况，就无法优选出站节点。总得来说observatory就是用来筛选优选出站的配置

**Reply by PatrickstarWritesCode** [reply]
I got the point, it's like a health check in load balancers, my question is what is the difference between observatory and burstobsservatory and their use cases.

**Reply by PatrickstarWritesCode** [reply]
Im currently using both since I dont know the difference

```
  "observatory": {
    "subjectSelector": [
      "Portal-1",
      "Portal-2",
      "Portal-3",
      "Portal-4",
      "Portal-5"
    ],
    "probeURL": "http://www.google.com/gen_204",
    "probeInterval": "10s",
    "enableConcurrency": true
  },
  "burstObservatory": {
    "subjectSelector": [
      "Portal-1",
      "Portal-2",
      "Portal-3",
      "Portal-4",
      "Portal-5"
    ],
    "pingConfig": {
      "destination": "https://www.google.com/gen_204",
      "interval": "1m",
      "connectivity": "http://connectivitycheck.platform.hicloud.com/generate_204",
      "timeout": "5s",
      "sampling": 5
    }
  }
```

**Reply by SatenRuiko-Lv0** [reply]
行吧，我英语确实不太好。翻译器翻译出来确实有点小问题。
leastPing的配置
"observatory": {
		"subjectSelector": ["global-out", "cn-out", "global-udp-out"],
		"probeURL": "http://www.apple.com/library/test/success.html",
		"probeInterval": "1h",
		"enableConcurrency": true
	}
leastload的配置
"burstObservatory": {
		"subjectSelector": ["global-out", "cn-out", "global-udp-out"],
		"pingConfig": {
            "destination": "http://www.apple.com/library/test/success.html",
            "interval": "1h",
            "connectivity": "http://connectivitycheck.platform.hicloud.com/generate_204",
            "timeout": "10s",
            "sampling": 2
        }
区别：leastping的observatory测的是不同出站之间的ping值，选取最小的那个出站。注意到"enableConcurrency": true了吗？通常情况下填false就行，意味着执行轮询检测，总体检测时长＝"probeInterval"乘以你一个策略组中的出站数量。填写true时则采用并发检测，一个"probeInterval"周期检测所有的ping优选延迟最小的出站。
leastload的burstObservatory则是检测每个出站之间的rtt值，优选出最小的那个选择出站。这种检测方式就是固定并发检测。destination是检测出站的rtt值。connectivity是检测你本地网络的连通性，如果本地网络有问题，则不会对出站节点的评估减分。sampling应该是保留最近几次检测结果吧？（大概）

**Reply by SatenRuiko-Lv0** [reply]
还有，ping只能用observatory，load只能用burstObservatory，强烈建议写一个fallback，不然当(burst)observatory失败时，会出现类似路由配置全部失效，所有流量默认走配置的第一个出站的情况。目前ping的fallback暂时不支持，据说作者会在下一个版本更新加进去。

**Reply by PatrickstarWritesCode** [reply]
Thank you for spending time on this. Really appreciated
